source/utilities.tex

```diff
@@ -21135,3 +21135,683 @@
+The class \tcode{stacktrace_entry} models
+\tcode{regular}\iref{concepts.object} and
+\tcode{three_way_comparable<strong_ordering>}\iref{cmp.concept}.
+
+\rSec3[stacktrace.entry.ctor]{Constructors}
+
+\begin{itemdecl}
+constexpr stacktrace_entry() noexcept;
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\ensures
+\tcode{*this} is empty.
+\end{itemdescr}
+
+\rSec3[stacktrace.entry.obs]{Observers}
+
+\begin{itemdecl}
+constexpr native_handle_type native_handle() const noexcept;
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+The semantics of this function are
+\impldef{semantics of \tcode{stacktrace_entry::native_handle}}.
+
+\pnum
+\remarks
+Successive invocations of the \tcode{native_handle} function
+for an unchanged \tcode{stacktrace_entry} object return identical values.
+\end{itemdescr}
+
+\begin{itemdecl}
+constexpr explicit operator bool() const noexcept;
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\returns
+\tcode{false} if and only if \tcode{*this} is empty.
+\end{itemdescr}
+
+\rSec3[stacktrace.entry.query]{Query}
+
+\pnum
+\begin{note}
+All the \tcode{stacktrace_entry} query functions treat
+errors other than memory allocation errors
+as ``no information available'' and do not throw in that case.
+\end{note}
+
+\begin{itemdecl}
+string description() const;
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\returns
+A description of the evaluation represented by \tcode{*this},
+or an empty string.
+
+\pnum
+\throws
+\tcode{bad_alloc} if memory for
+the internal data structures or the resulting string cannot be allocated.
+\end{itemdescr}
+
+\begin{itemdecl}
+string source_file() const;
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\returns
+The presumed or actual name of the source file\iref{cpp.predefined}
+that lexically contains the expression or statement
+whose evaluation is represented by \tcode{*this}, or an empty string.
+
+\pnum
+\throws
+\tcode{bad_alloc} if memory for
+the internal data structures or the resulting string cannot be allocated.
+\end{itemdescr}
+
+\begin{itemdecl}
+uint_least32_t source_line() const;
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\returns
+0, or a 1-based line number that lexically relates to the evaluation
+represented by \tcode{*this}.
+If \tcode{source_file} returns the presumed name of the source file,
+returns the presumed line number;
+if \tcode{source_file} returns the actual name of the source file,
+returns the actual line number.
+
+\pnum
+\throws
+\tcode{bad_alloc} if memory for
+the internal data structures or the resulting string cannot be allocated.
+\end{itemdescr}
+
+\rSec3[stacktrace.entry.cmp]{Comparison}
+
+\begin{itemdecl}
+friend constexpr bool operator==(const stacktrace_entry& x, const stacktrace_entry& y) noexcept;
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\returns
+\tcode{true} if and only if \tcode{x} and \tcode{y} represent
+the same stacktrace entry or both \tcode{x} and \tcode{y} are empty.
+\end{itemdescr}
+
+\rSec2[stacktrace.basic]{Class template \tcode{basic_stacktrace}}
```

Comment:
Shouldn't this be [basic.stacktrace]? The few class templates with basic in their stable names are ordered like that (`basic_string` and `basic_ios`).

Reply:
Counterexample: [string.view.template]

I'll leave the hierarchical ordering as-is; we're in a subclause "stacktrace", so all stable names start with "stacktrace".

Reply:
@tkoeppe , any opinion here?

Reply:
We don't have a policy on this, and I see no reason to change the approved wording here. It seems more useful to have the labels form a regular hierarchy than reflect class names.